odk1-src/form-functions.rst

```diff
@@ -0,0 +1,719 @@
+
+
+.. function:: substr(string, start[, end]) 	
+
+  Returns the substring of :arg:`string` beginning at the index :arg:`start` and extending to (but not including) index :arg:`end` (or to the termination of :arg:`string`, if :arg:`end` is not provided). Members of arg:`string` are zero-indexed.
+  
+.. function:: string-length(string)
+
+  Returns the number of characters in :arg:`string`.
+  
+.. _string-combination-functions:
+  
+Combining strings
+~~~~~~~~~~~~~~~~~~  
+
+.. function:: concat(arg [, arg [, arg [, arg [...]]]])
+
+  Concatenates one or more arguments into a single string. If any :arg:`arg` is a :term:`nodeset`, the values within the set are concatenated into a string.
+
+  
+.. function:: join(separator, nodeset)
+
+  Joins the members of :arg:`nodeset`, using the string :arg:`separator`.
+
+.. _string-conversion-functions:
+  
+Converting to and from strings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. function:: boolean-from-string(string)
+
+  Returns ``True`` if :arg:`string` is "true" or "1".
+  Otherwise, ``False``.
+
+
+.. function:: number(string)
+
+  Converts a :arg:`string` of digits into a number value.
+
+.. function:: string(arg)
+
+   Converts :arg:`arg` to a string.
+
+.. _math-functions:
+  
+Math 
+------
+
+.. _number-functions:
+
+Number handling
+~~~~~~~~~~~~~~~~~
+
+.. function:: round(number, places)
+
+  Rounds a decimal :arg:`number` to some number of decimal :arg:`places`.
+
+.. function:: int(arg) 	
+
+  Converts :arg:`arg` to an integer.
+
+.. seealso:: :func:`count`, :func:`max`, :func:`min`, :func:`number`
+  
+.. _calculation-functions:
+  
+Calculation
+~~~~~~~~~~~~~
+
+.. function:: pow(number, power)
+
+  Raises a :arg:`number` to a :arg:`power`.
+
+.. function:: log(number)
+
+  Returns the natural log of :arg:`number`.
+
+.. function:: log10(number)
+
+  Returns the base-10 log of :arg:`number`.
+
+.. function:: abs(number)
+
+  Returns the absolute value of :arg:`number`.
+
+.. function:: sin(number)
+
+  Returns the sine of :arg:`number`.
+
+.. function:: cos(number)
+
+  Returns the cosine of :arg:`number`.
+  
+.. function:: tan(number)
+
+  Returns the tangent of :arg:`number`.
+
+.. function:: asin(number)
+
+  Returns the arc sine of :arg:`number`.
+  
+.. function:: acos(number)
+
+  Returns the arc cosine of :arg:`number`.
+
+.. function:: atan(number)
+
+  Returns the arctan of :arg:`number`.
+
+.. function:: atan2(y,x)
+
+  Returns the multi-valued inverse tangent of :arg:`y`, :arg:`x`.
+
+.. function:: sqrt(number) 
+
+  Returns the square root of :arg:`number`.
+
+.. function:: exp(x) 
+
+  Returns ``e^x``.
+
+.. function:: exp10(x)
+
+  Returns ``10^x``.
+
+.. function:: pi()
+
+  Returns an approximation of the mathematical constant π.
+
+  
+.. _date-time-functions:
+    
+Date and time
+----------------
+
+.. function:: today()
+
+  Returns the current date without a time component.
+
+.. function:: now()
+
+  Returns the current datetime in the current time zone.
```

Comment:
in iso 8601 format including the timezone - https://en.wikipedia.org/wiki/ISO_8601